src/nodes/NodeManager.ts

```diff
@@ -573,169 +641,153 @@ class NodeManager {
       bucketIndex,
     );
     // We then need to start a findNode procedure
-    await this.nodeConnectionManager.findNode(bucketRandomNodeId, { signal });
+    await this.nodeConnectionManager.findNode(bucketRandomNodeId, true, {
+      signal,
+    });
   }
 
-  // Refresh bucket activity timer methods
+  private async setupRefreshBucketTasks(tran?: DBTransaction) {
+    if (tran == null) {
+      return this.db.withTransactionF((tran) =>
+        this.setupRefreshBucketTasks(tran),
+      );
+    }
 
-  private startRefreshBucketTimers() {
-    // Setting initial bucket to refresh
-    this.refreshBucketNext = 0;
-    // Setting initial deadline
-    this.refreshBucketTimerReset(this.refreshBucketTimerDefault);
+    this.logger.info('Setting up refreshBucket tasks');
+    // 1. Iterate over existing tasks and reset the delay
+    const existingTasks: Array<boolean> = new Array(this.nodeGraph.nodeIdBits);
+    for await (const task of this.taskManager.getTasks(
+      'asc',
+      true,
+      ['refreshBucket'],
+      tran,
+    )) {
+      const bucketIndex = parseInt(task.path[0]);
+      switch (task.status) {
+        case 'scheduled':
+          {
+            // If it's scheduled then reset delay
+            existingTasks[bucketIndex] = true;
+            // Total delay is refreshBucketDelay + time since task creation
+            const spread =
+              (Math.random() - 0.5) *
+              this.refreshBucketDelay *
+              this.refreshBucketDelaySpread;
+            const delay =
+              performance.now() +
+              performance.timeOrigin -
+              task.created.getTime() +
+              this.refreshBucketDelay +
+              spread;
+            await this.taskManager.updateTask(task.id, { delay }, tran);
+          }
+          break;
+        case 'queued':
+        case 'active':
+          // If it's running then leave it
+          existingTasks[bucketIndex] = true;
+          break;
+        default:
+          // Otherwise, ignore it, should be re-created
+          existingTasks[bucketIndex] = false;
+      }
+      this.logger.info('Set up refreshBucket tasks');
+    }
 
+    // 2. Recreate any missing tasks for buckets
     for (
       let bucketIndex = 0;
-      bucketIndex < this.nodeGraph.nodeIdBits;
+      bucketIndex < existingTasks.length;
       bucketIndex++
     ) {
-      const deadline = Date.now() + this.refreshBucketTimerDefault;
-      this.refreshBucketDeadlineMap.set(bucketIndex, deadline);
+      const exists = existingTasks[bucketIndex];
+      if (!exists) {
+        // Create a new task
+        this.logger.debug(
+          `Creating refreshBucket task for bucket ${bucketIndex}`,
+        );
+        const spread =
+          (Math.random() - 0.5) *
+          this.refreshBucketDelay *
+          this.refreshBucketDelaySpread;
```

Comment:
Can this be a utility function. Call it `nodesUtils.refreshBucketsDelayJitter`. It should a function that can give you this number that you adding to the `delay`.

Reply:
I have extracted this into the utils.